yt/chyt/controller/internal/agent/agent.go

```diff
@@ -409,10 +409,70 @@ func (a *Agent) Start() {
 		a.OperationNamespace(),
 		a.l)
 
+	go a.runScaler()
+
 	go a.background()
 	a.l.Info("agent started")
 }
 
+func (a *Agent) runScaler() {
+	scalePeriod := time.Duration(a.config.ScalePeriodOrDefault())
+	ticker := time.NewTicker(scalePeriod)
+
+	for {
+		select {
+		case <-a.ctx.Done():
+			return
+		case <-ticker.C:
+			startedAt := time.Now()
+
+			a.l.Info("Starting scaler")
+
+			workerNumber := a.config.ScaleWorkerNumberOrDefault()
+			var wg sync.WaitGroup
+			wg.Add(workerNumber)
+
+			opletsChan := make(chan *strawberry.Oplet, len(a.aliasToOp))
+
+			for i := 0; i < workerNumber; i++ {
+				go func() {
+					defer wg.Done()
+					for oplet := range opletsChan {
+						needsScaling, target, err := oplet.NeedsScaling(context.Background())
+						if err != nil {
+							a.l.Errorf("Got error processing %s: %v", oplet.Identificator(), err)
+							continue
+						}
+						if needsScaling {
+							a.l.Infof(
+								"Scaling %s -> %d",
+								oplet.Identificator(),
+								target.InstanceCount,
+							)
+							if target.InstanceCount == 0 {
+								err = oplet.SuspendOp(context.Background(), target.Reason)
+								if err != nil {
+									a.l.Errorf("Failed to suspend %s: %v", oplet.Identificator(), err)
+								}
+							} else {
+								a.l.Error("Not implemented")
```

Comment:
Let's note that it is scaling for a non-zero number of instances that is not implemented